src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorResolver.cs

```diff
@@ -256,23 +260,40 @@ private static bool MatchesLookupInfo(TagHelperDescriptor descriptor, LookupInfo
             {
                 errorSink.OnError(
                     directiveDescriptor.Location,
-                    Resources.FormatTagHelperDescriptorResolver_InvalidTagHelperLookupText(lookupText));
+                    Resources.FormatTagHelperDescriptorResolver_InvalidTagHelperLookupText(lookupText),
+                    GetErrorLength(lookupText));
 
                 return null;
             }
 
+            var trimmedAssemblyName = lookupStrings[1].Trim();
+            var assemblyNameIndex = lookupStrings[0].Length + 1 + lookupStrings[1].IndexOf(trimmedAssemblyName);
+            var assemblyNamePrefix = directiveDescriptor.DirectiveText.Substring(0, assemblyNameIndex);
+            var assemblyNameLocation = SourceLocation.Advance(directiveDescriptor.Location, assemblyNamePrefix);
+
             return new LookupInfo
             {
                 TypePattern = lookupStrings[0].Trim(),
-                AssemblyName = lookupStrings[1].Trim()
+                AssemblyName = trimmedAssemblyName,
+                AssemblyNameLocation = assemblyNameLocation,
             };
         }
 
+        private static int GetErrorLength(string directiveText)
+        {
+            var nonNullLength = directiveText?.Length ?? 1;
```

Comment:
Let's not box: if `var nonNullLength = directiveText != null ? directiveText.Length : 1;`